server/planning/types/locations.py

```diff
@@ -0,0 +1,78 @@
+from datetime import datetime
+from typing import Any, Annotated
+from pydantic import Field
+
+from planning.types.base import BasePlanningModel
+
+from superdesk.utc import utcnow
+from superdesk.core.resources import fields, dataclass
+from superdesk.core.resources.validators import validate_data_relation_async, validate_unique_value_async
+
+
+@dataclass
+class Position:
+    latitude: float | None = None
+    longitude: float | None = None
+    altitude: int | None = None
+    gps_datum: str | None = None
+
+
+@dataclass
+class Address:
+    title: str | None = None
+    line: list[str] | None = None
+    suburb: str | None = None
+    city: str | None = None
+    state: str | None = None
+    postal_code: str | None = None
+    country: str | None = None
+    locality: str | None = None  # Aka city
+    area: str | None = None
+    external: dict | None = None
+    boundingbox: list | None = None
+    address_type: str | None = Field(alias="type", default=None)
+
+
+class LocationResourceModel(BasePlanningModel):
+    guid: str | None = None
```

Comment:
As this resource uses Elasticsearch, we must ensure that the elastic mapping is correct.
To this end, anything in the old schema that has `"mapping": not_analyzed` needs to be replaced with [fields.Keyword](https://github.com/superdesk/superdesk-core/blob/async/superdesk/core/resources/fields.py#L126)

Reply:
Also any field that has `"unique": True` in the old schema required the [validate_iunique_value_async](https://github.com/superdesk/superdesk-core/blob/async/superdesk/core/resources/validators.py#L230) annotated for that field